inject.go

```diff
@@ -69,33 +85,44 @@ func Inject(target interface{}) Option {
 	// 		target.Field(2),  // Baz io.Writer
 	// 	]
 	//
-	// As we iterate through the arguments received by the function, we can
-	// simply copy the value into the corresponding value in the targets list.
+	// As we iterate through the fields of the generated struct, we can simply
+	// copy the value into the corresponding value in the targets list.
 	targets := make([]reflect.Value, 0, t.NumField())
 
 	for i := 0; i < t.NumField(); i++ {
 		f := t.Field(i)
-		// Skip private fields.
-		if f.PkgPath != "" {
+
+		// Skip private fields
+		if f.PkgPath != "" && !f.Anonymous {
```

Comment:
Interesting. There appears to be a slight difference between pointer and non-pointer embedding.

Given,

```go
type foo struct{ type1 }
```

`foo.type1` is an unexported field with `PkgPath != ""` and `Name == "type1"`.

However, given,

```go
type foo struct{ *type1 }
```

`foo.type1` is a field with `PkgPath == ""` and `Name == "type1"`. You *can*
use `reflect` to `Set` it.

This may be a bug in the `reflect` package.